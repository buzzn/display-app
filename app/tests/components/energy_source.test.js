import React from 'react';
import renderer from 'react-test-renderer';
import EnergySource from '../../components/energy_source';

describe('energy source component tests', () => {
  test('renders all energy types correctly', () => {
    const solarTree = renderer.create(
      <EnergySource type="solar"/>,
    ).toJSON();
    expect(solarTree).toMatchSnapshot();

    const fireTree = renderer.create(
      <EnergySource type="fire"/>,
    ).toJSON();
    expect(fireTree).toMatchSnapshot();

    const gridTree = renderer.create(
      <EnergySource type="grid"/>,
    ).toJSON();
    expect(gridTree).toMatchSnapshot();

    const consumptionTree = renderer.create(
      <EnergySource type="consumption"/>,
    ).toJSON();
    expect(consumptionTree).toMatchSnapshot();

    const prodStatTree = renderer.create(
      <EnergySource type="prodStat"/>,
    ).toJSON();
    expect(prodStatTree).toMatchSnapshot();

    const consStatTree = renderer.create(
      <EnergySource type="consStat"/>,
    ).toJSON();
    expect(consStatTree).toMatchSnapshot();

    const autarchyTree = renderer.create(
      <EnergySource type="autarchy"/>,
    ).toJSON();
    expect(autarchyTree).toMatchSnapshot();
  });

  test('renders left/right positions correctly', () => {
    const leftTree = renderer.create(
      <EnergySource type="solar" position="left"/>,
    ).toJSON();
    expect(leftTree).toMatchSnapshot();

    const rightTree = renderer.create(
      <EnergySource type="solar" position="right"/>,
    ).toJSON();
    expect(rightTree).toMatchSnapshot();
  });

  test('renders stat and autarchy values correctly', () => {
    const prodStatTree = renderer.create(
      <EnergySource type="prodStat" value={ 1000 }/>,
    ).toJSON();
    expect(prodStatTree).toMatchSnapshot();

    const consStatTree = renderer.create(
      <EnergySource type="constStat" value={ 1000 }/>,
    ).toJSON();
    expect(consStatTree).toMatchSnapshot();

    const autarchyTree = renderer.create(
      <EnergySource type="autarchy" value={ 0.88 }/>,
    ).toJSON();
    expect(autarchyTree).toMatchSnapshot();
  });
});
