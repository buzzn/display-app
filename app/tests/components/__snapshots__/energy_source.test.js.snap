// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`energy source component tests renders all energy types correctly 1`] = `
<div
  style={
    Object {
      "backgroundImage": "linear-gradient(to left, #ffffff, #ffeb3b)",
      "height": "120px",
      "marginBottom": "20px",
      "position": "relative",
      "width": "460px",
    }
  }
>
  <i
    className="fa fa-5x fa-sun-o"
    style={
      Object {
        "color": "#9e9e9e",
        "float": "left",
        "marginLeft": "32px",
        "marginTop": "25px",
        "minWidth": "80px",
      }
    }
  />
  <div
    className="power"
    style={
      Object {
        "float": "left",
        "fontSize": "34px",
        "marginLeft": "60px",
        "marginTop": "40px",
      }
    }
  >
    0 W
  </div>
  <div
    style={
      Object {
        "background": "white",
        "borderRadius": "60px",
        "height": "120px",
        "position": "absolute",
        "right": "-60px",
        "top": "0",
        "width": "120px",
      }
    }
  >
    <i
      className="fa fa-square"
      style={
        Object {
          "marginLeft": "54px",
          "marginTop": "50px",
        }
      }
    />
  </div>
</div>
`;

exports[`energy source component tests renders all energy types correctly 2`] = `
<div
  style={
    Object {
      "backgroundImage": "linear-gradient(to left, #ffffff, #ffa726)",
      "height": "120px",
      "marginBottom": "20px",
      "position": "relative",
      "width": "460px",
    }
  }
>
  <i
    className="fa fa-5x fa-fire"
    style={
      Object {
        "color": "#9e9e9e",
        "float": "left",
        "marginLeft": "32px",
        "marginTop": "25px",
        "minWidth": "80px",
      }
    }
  />
  <div
    className="power"
    style={
      Object {
        "float": "left",
        "fontSize": "34px",
        "marginLeft": "60px",
        "marginTop": "40px",
      }
    }
  >
    0 W
  </div>
  <div
    style={
      Object {
        "background": "white",
        "borderRadius": "60px",
        "height": "120px",
        "position": "absolute",
        "right": "-60px",
        "top": "0",
        "width": "120px",
      }
    }
  >
    <i
      className="fa fa-square"
      style={
        Object {
          "marginLeft": "54px",
          "marginTop": "50px",
        }
      }
    />
  </div>
</div>
`;

exports[`energy source component tests renders all energy types correctly 3`] = `
<div
  style={
    Object {
      "backgroundImage": "linear-gradient(to left, #ffffff, #bdbdbd)",
      "height": "120px",
      "marginBottom": "20px",
      "position": "relative",
      "width": "460px",
    }
  }
>
  <div
    style={
      Object {
        "color": "#9e9e9e",
        "float": "left",
        "marginLeft": "32px",
        "marginTop": "19px",
        "minWidth": "80px",
      }
    }
  >
    <svg
      height="80"
      version="1.1"
      viewBox="0 0 565 687"
      width="80"
    >
      <g
        fill="none"
        fillRule="evenodd"
        id="Page-1"
        stroke="none"
        strokeWidth="1"
      >
        <g
          fill="#9e9e9e"
          fillRule="nonzero"
          id="noun_1053838_cc"
          stroke="#9e9e9e"
          strokeWidth="10"
          transform="translate(5.000000, 5.000000)"
        >
          <path
            d="M554.4468,138.165286 L554.4468,137.272457 C554.4468,136.379628 553.553971,135.486799 552.661142,134.59397 C552.661142,134.59397 552.661142,134.59397 551.768313,134.59397 C550.875484,133.701141 550.875484,133.701141 549.982655,132.808312 L281.241131,0.66962174 C278.562644,-0.223207247 275.884157,-0.223207247 273.20567,0.66962174 L5.35697392,132.808312 C4.46414493,132.808312 3.57131594,133.701141 3.57131594,134.59397 C3.57131594,134.59397 3.57131594,134.59397 2.67848696,134.59397 C1.78565797,135.486799 0.892828986,136.379628 0.892828986,137.272457 L0.892828986,138.165286 C0.892828986,139.058115 0.892828986,139.950944 0,140.843773 L0,140.843773 L0,198.877657 C0,204.234631 3.57131594,207.805947 8.92828986,207.805947 C14.2852638,207.805947 17.8565797,204.234631 17.8565797,198.877657 L17.8565797,149.772062 L156.245073,149.772062 L156.245073,231.0195 L5.35697392,305.124306 C4.46414493,305.124306 3.57131594,306.017135 3.57131594,306.909964 C3.57131594,306.909964 3.57131594,306.909964 2.67848696,306.909964 C1.78565797,307.802793 0.892828986,308.695622 0.892828986,309.588451 L0.892828986,310.48128 C0.892828986,311.374109 0.892828986,312.266938 0,313.159767 L0,313.159767 L0,371.193651 C0,376.550625 3.57131594,380.121941 8.92828986,380.121941 C14.2852638,380.121941 17.8565797,376.550625 17.8565797,371.193651 L17.8565797,322.088057 L156.245073,322.088057 L156.245073,401.549837 L72.3191479,664.041558 C71.4263189,667.612874 72.3191479,672.077019 75.8904638,673.862677 C77.6761218,674.755506 79.4617798,675.648335 80.3546088,675.648335 C82.1402667,675.648335 84.8187537,674.755506 86.6044117,673.862677 L274.098499,508.689315 L463.378244,673.862677 C466.056731,676.541164 470.520876,676.541164 474.092192,674.755506 C477.663508,672.969848 478.556337,668.505703 477.663508,664.934387 L392.844754,402.442666 L392.844754,322.980886 L536.590221,322.980886 L536.590221,372.08648 C536.590221,377.443454 540.161537,381.01477 545.518511,381.01477 C550.875484,381.01477 554.4468,377.443454 554.4468,372.08648 L554.4468,314.052596 L554.4468,314.052596 C554.4468,313.159767 554.4468,312.266938 553.553971,311.374109 L553.553971,310.48128 C553.553971,309.588451 552.661142,308.695622 551.768313,307.802793 C551.768313,307.802793 551.768313,307.802793 550.875484,307.802793 C549.982655,306.909964 549.982655,306.909964 549.089826,306.017135 L391.951925,229.233842 L391.951925,150.664891 L535.697392,150.664891 L535.697392,199.770486 C535.697392,205.12746 539.268708,208.698776 544.625682,208.698776 C549.982655,208.698776 553.553971,205.12746 553.553971,199.770486 L553.553971,141.736602 L553.553971,141.736602 C554.4468,139.950944 554.4468,139.058115 554.4468,138.165286 Z M277.669815,18.5262015 L374.988174,66.7389667 L374.988174,131.915483 L174.994481,131.915483 L174.994481,69.4174537 L277.669815,18.5262015 Z M48.2127653,131.915483 L157.137902,78.3457435 L157.137902,131.915483 L48.2127653,131.915483 Z M48.2127653,304.231477 L157.137902,250.661738 L157.137902,304.231477 L48.2127653,304.231477 Z M174.994481,241.733448 L277.669815,190.842196 L374.988174,239.054961 L374.988174,304.231477 L174.994481,304.231477 L174.994481,241.733448 Z M169.637507,418.513587 L259.813235,496.189709 L99.1040175,638.149518 L169.637507,418.513587 Z M450.878638,639.042347 L287.490934,496.189709 L379.452319,418.513587 L450.878638,639.042347 Z M374.988174,398.87135 L273.20567,484.582932 L174.101652,398.87135 L174.101652,322.088057 L374.988174,322.088057 L374.988174,398.87135 Z M508.019693,304.231477 L392.844754,304.231477 L392.844754,247.983251 L508.019693,304.231477 Z M374.988174,219.412723 L281.241131,172.985616 C278.562644,172.092787 275.884157,172.092787 273.20567,172.985616 L174.994481,222.09121 L174.994481,149.772062 L375.881003,149.772062 L375.881003,219.412723 L374.988174,219.412723 Z M392.844754,131.915483 L392.844754,75.6672566 L507.126864,131.915483 L392.844754,131.915483 Z"
            id="Shape"
          />
        </g>
      </g>
    </svg>
  </div>
  <div
    className="power"
    style={
      Object {
        "float": "left",
        "fontSize": "34px",
        "marginLeft": "60px",
        "marginTop": "40px",
      }
    }
  >
    0 W
  </div>
  <div
    style={
      Object {
        "background": "white",
        "borderRadius": "60px",
        "height": "120px",
        "position": "absolute",
        "right": "-60px",
        "top": "0",
        "width": "120px",
      }
    }
  >
    <i
      className="fa fa-square"
      style={
        Object {
          "marginLeft": "54px",
          "marginTop": "50px",
        }
      }
    />
  </div>
</div>
`;

exports[`energy source component tests renders all energy types correctly 4`] = `
<div
  style={
    Object {
      "backgroundImage": "linear-gradient(to left, #ffffff, #80deea)",
      "height": "120px",
      "marginBottom": "20px",
      "position": "relative",
      "width": "460px",
    }
  }
>
  <div
    className="power"
    style={
      Object {
        "fontSize": "34px",
        "paddingTop": "20px",
        "textAlign": "center",
        "width": "100%",
      }
    }
  >
    <div>
      0 W
    </div>
    <div
      style={
        Object {
          "fontSize": "20px",
        }
      }
    >
      Aktueller Verbrauch
    </div>
  </div>
</div>
`;

exports[`energy source component tests renders all energy types correctly 5`] = `
<div
  style={
    Object {
      "backgroundImage": "linear-gradient(to left, #ffffff, #d4e157)",
      "height": "120px",
      "marginBottom": "20px",
      "position": "relative",
      "width": "460px",
    }
  }
>
  <div
    className="power"
    style={
      Object {
        "fontSize": "34px",
        "paddingTop": "20px",
        "textAlign": "center",
        "width": "100%",
      }
    }
  >
    <div>
      n.a.
    </div>
    <div
      style={
        Object {
          "fontSize": "20px",
        }
      }
    >
      Tagesproduktion
    </div>
  </div>
</div>
`;

exports[`energy source component tests renders all energy types correctly 6`] = `
<div
  style={
    Object {
      "backgroundImage": "linear-gradient(to left, #ffffff, #80deea)",
      "height": "120px",
      "marginBottom": "20px",
      "position": "relative",
      "width": "460px",
    }
  }
>
  <div
    className="power"
    style={
      Object {
        "fontSize": "34px",
        "paddingTop": "20px",
        "textAlign": "center",
        "width": "100%",
      }
    }
  >
    <div>
      n.a.
    </div>
    <div
      style={
        Object {
          "fontSize": "20px",
        }
      }
    >
      Tagesverbrauch
    </div>
  </div>
</div>
`;

exports[`energy source component tests renders all energy types correctly 7`] = `
<div
  style={
    Object {
      "backgroundImage": "linear-gradient(to left, #ffffff, #bdbdbd)",
      "height": "120px",
      "marginBottom": "20px",
      "position": "relative",
      "width": "460px",
    }
  }
>
  <div
    className="power"
    style={
      Object {
        "fontSize": "34px",
        "paddingTop": "20px",
        "textAlign": "center",
        "width": "100%",
      }
    }
  >
    <div>
      n.a.
    </div>
    <div
      style={
        Object {
          "fontSize": "20px",
        }
      }
    >
      Autarkie heute
    </div>
  </div>
</div>
`;

exports[`energy source component tests renders left/right positions correctly 1`] = `
<div
  style={
    Object {
      "backgroundImage": "linear-gradient(to left, #ffffff, #ffeb3b)",
      "height": "120px",
      "marginBottom": "20px",
      "position": "relative",
      "width": "460px",
    }
  }
>
  <i
    className="fa fa-5x fa-sun-o"
    style={
      Object {
        "color": "#9e9e9e",
        "float": "left",
        "marginLeft": "32px",
        "marginTop": "25px",
        "minWidth": "80px",
      }
    }
  />
  <div
    className="power"
    style={
      Object {
        "float": "left",
        "fontSize": "34px",
        "marginLeft": "60px",
        "marginTop": "40px",
      }
    }
  >
    0 W
  </div>
  <div
    style={
      Object {
        "background": "white",
        "borderRadius": "60px",
        "height": "120px",
        "position": "absolute",
        "right": "-60px",
        "top": "0",
        "width": "120px",
      }
    }
  >
    <i
      className="fa fa-square"
      style={
        Object {
          "marginLeft": "54px",
          "marginTop": "50px",
        }
      }
    />
  </div>
</div>
`;

exports[`energy source component tests renders left/right positions correctly 2`] = `
<div
  style={
    Object {
      "backgroundImage": "linear-gradient(to right, #ffffff, #ffeb3b)",
      "height": "120px",
      "marginBottom": "20px",
      "position": "relative",
      "width": "460px",
    }
  }
>
  <i
    className="fa fa-5x fa-sun-o"
    style={
      Object {
        "color": "#9e9e9e",
        "float": "right",
        "marginRight": "32px",
        "marginTop": "25px",
        "minWidth": "80px",
      }
    }
  />
  <div
    className="power"
    style={
      Object {
        "float": "right",
        "fontSize": "34px",
        "marginRight": "60px",
        "marginTop": "40px",
      }
    }
  >
    0 W
  </div>
  <div
    style={
      Object {
        "background": "white",
        "borderRadius": "60px",
        "height": "120px",
        "left": "-60px",
        "position": "absolute",
        "top": "0",
        "width": "120px",
      }
    }
  >
    <i
      className="fa fa-square"
      style={
        Object {
          "marginLeft": "54px",
          "marginTop": "50px",
        }
      }
    />
  </div>
</div>
`;

exports[`energy source component tests renders stat and autarchy values correctly 1`] = `
<div
  style={
    Object {
      "backgroundImage": "linear-gradient(to left, #ffffff, #d4e157)",
      "height": "120px",
      "marginBottom": "20px",
      "position": "relative",
      "width": "460px",
    }
  }
>
  <div
    className="power"
    style={
      Object {
        "fontSize": "34px",
        "paddingTop": "20px",
        "textAlign": "center",
        "width": "100%",
      }
    }
  >
    <div>
      1 kWh
    </div>
    <div
      style={
        Object {
          "fontSize": "20px",
        }
      }
    >
      Tagesproduktion
    </div>
  </div>
</div>
`;

exports[`energy source component tests renders stat and autarchy values correctly 2`] = `
<div
  style={
    Object {
      "backgroundImage": "linear-gradient(to left, #ffffff, #ffffff)",
      "height": "120px",
      "marginBottom": "20px",
      "position": "relative",
      "width": "460px",
    }
  }
>
  <i
    className="fa fa-5x "
    style={
      Object {
        "color": "#9e9e9e",
        "float": "left",
        "marginLeft": "32px",
        "marginTop": "25px",
        "minWidth": "80px",
      }
    }
  />
  <div
    className="power"
    style={
      Object {
        "float": "left",
        "fontSize": "34px",
        "marginLeft": "60px",
        "marginTop": "40px",
      }
    }
  >
    1 kW
  </div>
  <div
    style={
      Object {
        "background": "white",
        "borderRadius": "60px",
        "height": "120px",
        "position": "absolute",
        "right": "-60px",
        "top": "0",
        "width": "120px",
      }
    }
  >
    <i
      className="fa fa-4x fa-chevron-left"
      style={
        Object {
          "marginLeft": "40px",
          "marginTop": "30px",
        }
      }
    />
  </div>
</div>
`;

exports[`energy source component tests renders stat and autarchy values correctly 3`] = `
<div
  style={
    Object {
      "backgroundImage": "linear-gradient(to left, #ffffff, #bdbdbd)",
      "height": "120px",
      "marginBottom": "20px",
      "position": "relative",
      "width": "460px",
    }
  }
>
  <div
    className="power"
    style={
      Object {
        "fontSize": "34px",
        "paddingTop": "20px",
        "textAlign": "center",
        "width": "100%",
      }
    }
  >
    <div>
      88%
    </div>
    <div
      style={
        Object {
          "fontSize": "20px",
        }
      }
    >
      Autarkie heute
    </div>
  </div>
</div>
`;
